frontend/src/components/referenceGenomes/ReferenceGenomeTableColumns.tsx

```diff
@@ -0,0 +1,120 @@
+
+export const getColumnsForReferenceGenome = (taxonsByID): ReferenceGenomeColumn[] => {
+    const REFERENCE_GENOME_COLUMN_DEFINITIONS = REFERENCE_GENOME_COLUMNS(taxonsByID)
+    return REFERENCE_GENOME_COLUMN_DEFINITIONS.map((column: ReferenceGenomeColumn) => { return { ...column } });
+}
+
+
+const REFERENCE_GENOME_COLUMNS = (taxonsByID): ReferenceGenomeColumn[] => [
+    {
+        columnID: ReferenceGenomeID.ID,
+        title: 'ID',
+        dataIndex: ['referenceGenome', 'id'],
+        sorter: (a, b) => a.id - b.id,
+        render: (_, { id }) =>
+            id && (
+                <Link to={`/genomes/update/${id}`}>
+                    <div>{id}</div>
+                </Link>
+            ),
+    },
+    {
+        columnID: ReferenceGenomeID.ASSEMBLY_NAME,
+        title: 'Assembly Name',
+        dataIndex: ['referenceGenome', 'assembly_name'],
+        sorter: (a, b) => a.assembly_name.localeCompare(b.assembly_name),
+        render: (_, { id, assembly_name }) =>
+            assembly_name && id && (
+                <Link to={`/genomes/update/${id}`}>
+                    <div>{assembly_name}</div>
+                </Link>
+            ),
+    },
+    {
+        columnID: ReferenceGenomeID.SYNONYM,
+        title: 'Synonym',
+        dataIndex: ['referenceGenome', 'synonym'],
+        sorter: (a, b) => {
+            if (a && b && a.synonym && b.synonym) {
+
+                return a.synonym.localeCompare(b.synonym);
+            }
+            return 0;
+        },
+        render: (_, { synonym }) =>
+            synonym && (
+                <div>{synonym}</div>
+            ),
+    },
+    {
+        columnID: ReferenceGenomeID.GENBANK_ID,
+        title: 'Genbank ID',
+        dataIndex: ['referenceGenome', 'genbank_id'],
+        sorter: (a, b) => {
+            if (a && b && a.genbank_id && b.genbank_id) {
+
+                return a.genbank_id.localeCompare(b.genbank_id);
+            }
+            return 0;
+        },
+        render: (_, { genbank_id }) =>
+            genbank_id && (
+                <div>{genbank_id}</div>
+            ),
+    },
+    {
+        columnID: ReferenceGenomeID.REFSEQ_ID,
+        title: 'Refseq ID',
+        dataIndex: ['referenceGenome', 'refseq_id'],
+        sorter: (a, b) => {
+            if (a && b && a.refseq_id && b.refseq_id) {
+
+                return a.refseq_id.localeCompare(b.refseq_id);
+            }
+            return 0;
+        },
+        render: (_, { refseq_id }) =>
+            refseq_id && (
+                <div>{refseq_id}</div>
+            ),
+    },
+    {
+        columnID: ReferenceGenomeID.TAXON_ID,
+        title: 'Taxon',
+        dataIndex: ['referenceGenome', 'taxon_id'],
+        sorter: (a, b) => a.taxon_id - b.taxon_id,
+        render: (_, { taxon_id }) =>
+            taxon_id && (
+                <Link to={`/taxons/update/${taxon_id}`}>
+                    <div>{taxonsByID[taxon_id].name}</div>
```

Comment:
Here it would be safer to use a "WithItemRenderComponent". That's a component that takes an object id and will fetch the object, if necessary. This code would bomb if the taxon with `taxon_id` doesn't exist. You will have to add a new `WithTaxonRenderComponent` for this in the file `WithItemRenderComponent.tsx`.
You won't need to pass the taxonsByID parameter to this function anymore, once you have a WithTaxonRenderComponent here.